kubetest/gke_test.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func newGkeDeployerFactory(instance_group_prefix string) (*gkeDeployer, error) {
+	flag.Set("gke-environment", "staging")
+	if instance_group_prefix != "" {
+		flag.Set("gke-instance-group-prefix", instance_group_prefix)
+	} else {
+		flag.Set("gke-instance-group-prefix", "gke")
+	}
+
+	return newGKE(
+		"gke",
+		"project",
+		"",
+		"region",
+		"network",
+		"image",
+		"imageFamily",
+		"imageProject",
+		"cluster",
+		"sshProxyInstanceName",
+		new(string),
+		new(string),
+	)
+}
+
+func TestParseInstanceGroupsFromGCloud(t *testing.T) {
```

Comment:
With  all those variables, I  have problems tracking what is being actually tested :-/

Why not use plain strings in those tests?

Reply:
I'm pretty sure that with the length of these strings it would be quite hard to read it. Variables' names kind of summarize what's important in string without reading whole string.

Reply:
I've replaces variables with strings.